src/main/java/com/microsoft/jenkins/kubernetes/ExtensionV1beta1ResourceManager.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.jenkins.kubernetes;
+
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.apis.ExtensionsV1beta1Api;
+import io.kubernetes.client.models.ExtensionsV1beta1Deployment;
+
+import java.io.IOException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class ExtensionV1beta1ResourceManager extends ResourceManager {
```

Comment:
why the class name contains version, will there be new class in future if new version released? suggest to make it general interface, then versioned instance as implementation of interface

Reply:
`**VersionResourceManager` just groups all the resourceUpdaters which actually do the job for each resource. And if a new k8s API version released, it needs a new `**ResourceManager` to keep added or updated `resourceUpdater`. This makes sure we do not break compatibility with existing clients and deal with the resources across different version. One example is Deployment resource occurs in both V1 and ExtensionV1beta1, so we need to add DeploymentUpdater in both `V1ResourceManger` and `ExtensionV1beta1ResourceManager`. `**ResourceManager` now just exposes one method `apply(T resource)` which handle related resources and this method is declared in abstract class `ResourceManager`.